src/Recorders/Queues.php

```diff
@@ -97,4 +105,30 @@ protected function getDefaultQueue(string $connection): string
     {
         return $this->config->get('queue.connections.'.$connection.'.queue', 'default');
     }
+
+    /**
+     * Normalize the SQS queue name.
+     */
+    protected function normalizeSqsQueue(string $connection, string $queue): string
+    {
+        $config = $this->config->get("queue.connections.{$connection}") ?? [];
+
+        if (($config['driver'] ?? null) !== 'sqs') {
+            return $connection;
+        }
+
+        if ($config['prefix'] ?? null) {
+            $prefix = preg_quote($config['prefix'], '#');
+
+            $queue = preg_replace("#^{$prefix}/#", '', $queue) ?? $queue;
+        }
+
+        if ($config['suffix'] ?? null) {
+            $suffix = preg_quote($config['suffix'], '#');
+
+            $queue = preg_replace("#{$suffix}$#", '', $queue) ?? $queue;
+        }
+
+        return $queue;
+    }
```

Comment:
Made this method naming SQS specific. If we have to do more normalizing we can introduce a "normalizeQueue" method that wraps other methods.

The `preg_replace(...) ?? $queue` null coalesce thingo is there because `preg_replace` can fail and return `null`.